designer/server/src/main/scala/pl/touk/nussknacker/engine/CombinedProcessingTypeData.scala

```diff
@@ -1,17 +1,21 @@
 package pl.touk.nussknacker.engine
 
 import pl.touk.nussknacker.restmodel.process.ProcessingType
-import pl.touk.nussknacker.ui.process.ProcessStateDefinitionService
+import pl.touk.nussknacker.ui.component.{ComponentIdProvider, DefaultComponentIdProvider}
+import pl.touk.nussknacker.ui.process.{ProcessCategoryService, ProcessStateDefinitionService}
 import pl.touk.nussknacker.ui.process.ProcessStateDefinitionService.StatusNameToStateDefinitionsMapping
 
 case class CombinedProcessingTypeData(statusNameToStateDefinitionsMapping: StatusNameToStateDefinitionsMapping,
+                                      componentIdProvider: ComponentIdProvider,
```

Comment:
Why not to keep here validated result of ComponentObjectsService.prepareWithoutFragments instead? It is the most valuable object that can be computed eagerly based on processing type: definitions of components after deduplication, enriched with componentId. WDYT?

Reply:
You're right but it would need further refactoring. Currently `UIProcessObjects` depend on fragments. In order to apply your suggestion we would need to be able to somehow enrich partial `UIProcessObjects` based only on standard components (services, sources, sinks) with fragments.

Reply:
Let's leave it on some future